AWSIoT/AWSIoTDataManager.h

```diff
@@ -416,6 +416,12 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)enableMetricsCollection:(BOOL)enabled;
 
+/**
+ Set user-specified dictionary of the additional values to be passed as components of
+ connection username.
+ */
+- (void)setUsernameFields:(NSDictionary<NSString *, NSString *> *)usernameFields;
```

Comment:
Lets see if we can come up with a different name for this api - something along the lines of "setUserNameMetaData"

Reply:
Done